### PR TITLE
ipsec: Refactor `IPSecDir`

### PR DESCRIPTION
**Pull request by pchaigno**
See commits. No functional changes.

### CODE REVIEW

**Review by jschwinger233 — APPROVED** [review]

**Comment by pchaigno** [comment]
Waiting for @ldelossa to merge his PR first as it conflicts with this one and is more painful to rebase.

File: pkg/datapath/linux/ipsec/ipsec_linux_test.go [hunk 1]
```diff
@@ -38,6 +38,10 @@ func setupIPSecSuitePrivileged(tb testing.TB) *slog.Logger {
 	return log
 }
 
+const (
+	ipsecDirBoth IPSecDir = IPSecDirIn | IPSecDirOut
```
Comment (ldelossa) [inline pkg/datapath/linux/ipsec/ipsec_linux_test.go]: Lint is correct here. No need for this anymore, the concept of 'both' does not exist.

**Review by ldelossa — APPROVED** [review]

File: pkg/datapath/linux/ipsec/ipsec_linux_test.go (hunk 1 above)
Comment (pchaigno) [reply pkg/datapath/linux/ipsec/ipsec_linux_test.go]: Oops, forgot to remove it.

**Pull request merged**